java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -386,14 +386,14 @@ private native void nativeCreateIndex(
    *
    * @return num of rows
    */
-  public int countRows() {
+  public long countRows() {
     try (LockManager.ReadLock readLock = lockManager.acquireReadLock()) {
       Preconditions.checkArgument(nativeDatasetHandle != 0, "Dataset is closed");
       return nativeCountRows();
     }
   }
 
-  private native int nativeCountRows();
+  private native long nativeCountRows();
```

Comment:
we do support countRows with filter. 
Should we implement this as `nativeCountRows(Optional<String> filter)` ?

Reply:
It's a good suggestion. We can add this method for the aggregator pushdown for spark connector. 
In this pr, the spark statistics only need the total row number. 
This statistic will be used in the optimizer. The `nativeCountRows(Optional<String> filter)` will use the cpu to do the filter, and it's not suitable for the statistics.

Reply:
I was thinking this as a Dataset public API aspect. we do need to expose the Dataset.java api as equivalent to the Rust Dataset API. 

In spark, it is fine to just call `nativeCountRows(None)` for planning.